frappe/public/js/frappe/router.js

```diff
@@ -469,19 +469,34 @@ frappe.router = {
 				return encodeURIComponent(String(a));
 			}
 		}).join("/");
-		let private_home = frappe.workspaces[`home-${frappe.user.name.toLowerCase()}`];
 
-		let workspace_name = private_home || frappe.workspaces["home"] ? "home" : "";
-		let is_private = !!private_home;
-		let first_workspace = Object.keys(frappe.workspaces)[0];
+		if (path_string) {
+			return "/app/" + path_string;
+		}
+
+		// Workspace
+		let private_home = `home-${frappe.user.name.toLowerCase()}`;
```

Comment:
This probably adds username twice, does it work for you?

Reply:
To be fair, I don't know what the original code does, I never knew there where private home workspaces. I tried to keep it unchanged during this refactor, maybe I made a mistake somewhere?

Reply:
![image](https://github.com/frappe/frappe/assets/9079960/ca1b0417-1da9-44cc-8c19-97e346cb60d5)


Getting this error, I have no idea how this works either. 

But:

![image](https://github.com/frappe/frappe/assets/9079960/8606d9b5-50fc-45fb-9caf-f8b2d196a615)

Reply:
I didn't anticipate that it was done **_again_** in `convert_to_standard_route`.

https://github.com/frappe/frappe/blob/caf7aec2865aad9f7f8e26fe17304baa0732b751/frappe/public/js/frappe/router.js#L180-L203

Reply:
Code would be cleaner if `convert_to_standard_route` didn't do shenanigans with private workspaces and localStorage :sweat_smile:

Reply:
Code would be cleaner if workspace name wasn't concoction of workspace + username. :smile: